include/zephyr/drivers/bluetooth/hci_driver.h

```diff
@@ -159,6 +159,24 @@ struct bt_hci_driver {
 	 */
 	uint32_t quirks;
 
+#if defined(CONFIG_BT_HCI_SET_PUBADDR) || defined(__DOXYGEN__)
+	/**
+	 * @brief Sets the controller's public address.
+	 *
+	 * Allows setting the controller's public address through
+	 * vendor-specific means. If the controller does not support changing
+	 * the public address, this should be NULL.
+	 *
+	 * This function is called before open().
+	 *
+	 * @note @kconfig{CONFIG_BT_HCI_SET_PUBADDR} must be selected for this
+	 * field to be available.
+	 *
+	 * @return 0 on success or negative error number on failure.
+	 */
+	int (*set_public_address)(bt_addr_t *addr);
```

Comment:
Name seems generally fine, but maybe we want to be consistent with shortening "address" to "addr" like in other Bluetooth APIs (e.g. `bt_addr_t`)?